src/Documentation/2.0/SiloShutdown.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: page
+title: Shutting down Orleans
+---
+
+This document explains how to gracefully shutdown an Orleans silo before application exit, first as a Console app, and then as a Docker container app.
+
+# Graceful shutdown - Console app
+The following code shows how to gracefully shutdown an orleans silo console app in response to the user pressing Ctrl+C, which generates the `Console.CancelkeyPress` event.
+
+Normally when that event handler returns, the application will exit immediately, causing a catastrophic Orleans silo crash and loss of data. But in the sample code below, we set `a.Cancel = true;` to prevent the application closing before the Orleans silo has completed its graceful shutdown.
+
+```csharp
+using Microsoft.Extensions.Logging;
+using Orleans.Configuration;
+using Orleans.Hosting;
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MySiloHost {
+
+    class Program {
+
+        static readonly ManualResetEvent _siloStarted = new ManualResetEvent(false);
+        static readonly ManualResetEvent _siloStopped = new ManualResetEvent(false);
+
+        static ISiloHost silo;
+        static bool siloStopping = false;
+        static readonly object syncLock = new object();
+
+        static void Main(string[] args) {
+
+            SetupApplicationShutdown();
+
+            silo = CreateSilo();
+            Task.Run(StartSilo);
+            _siloStarted.WaitOne();
+
+            /// Wait for the silo to completely shutdown before exiting. 
+            _siloStopped.WaitOne();
+        }
+
+        static void SetupApplicationShutdown() {
+            /// Capture the user pressing Ctrl+C
+            Console.CancelKeyPress += (s, a) => {
+                /// Prevent the application from crashing ungracefully.
+                a.Cancel = true;
+                /// Don't allow the following code to repeat if the user presses Ctrl+C repeatedly.
+                lock (syncLock) {
+                    if (!siloStopping) {
+                        siloStopping = true;
+                        Task.Run(StopSilo);
```

Comment:
Since the `Task` returned by `Task.Run()` isn't awaited, it's better to add `.Ignore()` to it, to make sure any exception it might throw will not get to the finalizer thread.